sdpb/api/stations.py

```diff
@@ -287,7 +299,90 @@ def collection(
 
 
 ####
-# /stations/{station_id}/observations/{var_id}
+# /stations/{station_id}/variables/{var_id}
+####
+
+
+def station_variable_timespan_query(session, station_id, var_id):
+    """
+    Returns a query with the minimum and maximum timestamps for
+    observations of this variable at this station. History-agnostic;
+    the minimum and maximum might correspond to different histories
+    if multiple histories record the same variable at one station.
+    """
+
+    return (
+        session.query(
+            func.min(VarsPerHistory.start_time).label("min_obs_time"),
+            func.max(VarsPerHistory.end_time).label("max_obs_time"))
+            .select_from(History)
+            .join(VarsPerHistory, VarsPerHistory.history_id == History.id)
+            .filter(History.station_id == station_id)
+            .filter(VarsPerHistory.vars_id == var_id)
+        )
+
+
+def get_station_variable(station_id, var_id):
+    """
+    Metadata about a variable specifically in the context of a station.
+    Returns the usual variable information from the /variable/{id} endpoint,
+    plus start and end times for data for this variable for this station.
+    """
+    assert station_id is not None, "station_id must be specified"
+    assert var_id is not None, "var_id must be specified"
+    session = get_app_session()
+    timespan = station_variable_timespan_query(session, station_id, var_id).one()
+
+    var = variables.single(var_id)
+    var["min_obs_time"] = timespan.min_obs_time
+    var["max_obs_time"] = timespan.max_obs_time
+    var["station_id"] = station_id
+    return var
+
+
+####
+# /stations/{station_id}/variables/
+####
+
+
+def get_station_variables(station_id=None):
+    assert station_id is not None
+    session = get_app_session()
+    q = session.query(Station).select_from(Station).filter(Station.id == station_id)
+    q = add_station_network_publish_filter(q)
+    station = q.one()
+    station_histories_etc = (
```

Comment:
This exact query is repeated in the function `single`. Suggest DRYing both up by extracting to a function.